birdhouse/pavics-compose.sh

```diff
@@ -25,10 +26,17 @@ VARS='
   $POSTGRES_PAVICS_PASSWORD
   $POSTGRES_MAGPIE_USERNAME
   $POSTGRES_MAGPIE_PASSWORD
+  $SERVER_NAME
+  $SERVER_SYNOPSIS
+  $SERVER_INSTITUTION
+  $SUPPORT_EMAIL
+  $SUPPORT_URL
+  $DATA_PERSIST_ROOT
 '
 
-# list of vars to be substituted in template but they do not have to be set in
-# env.local
+# list of vars to be substituted in template but they do not have to be set in env.local
+#   these variables can be omitted completely as they can be inferred from other values or are not specifically required
```

Comment:
I can reword it as suggested.

Reply:
I re-read your comment. 
> why not just say "their default values are from default.env so they do not have to be defined in env.local".

and 
> is misleading [...] Because they are required for real during template expansion

Both are incorrect. They are not defined in `default.env`, nor are *required* anywhere.
And, they really are variables that are computed one way or another from other values, which can be overridden for convenience. 

Best example, first one `PAVICS_FQDN_PUBLIC`, is defined only in `pavics-compose.sh` and, uses `PAVICS_FQDN` if not defined, and even uses another value in `configure-pavics.sh` when using Kite.

Reply:
> Both are incorrect. They are not defined in `default.env`, nor are _required_ anywhere.

So there are only 3 exceptions that are not in `default.env` and the default value computed in `pavics-compose.sh` directly.  They are not in `default.env` because their value needs to be computed from other values as you've said it yourself (ex: `PAVICS_FQDN_PUBLIC`).  But it's the same "have default value" spirit here.

They are also _required_ because they are used in template expansion, example `PAVICS_FQDN_PUBLIC`: https://github.com/bird-house/birdhouse-deploy/blob/57a320c3583f804838e8c18aeb064266527a0bc9/birdhouse/config/canarie-api/docker_configuration.py.template#L7

So what is "incorrect" other than the technicality where the default value comes from?

Reply:
I think that in a configuration point of view, for an external dev configuring a new platform, a variable that defaults to some value if not provided, regardless of how that resolution occurs, is not required. 
It would be even more misleading to say that `OPTIONAL_VARS` is required.

Reply:
> I think that in a configuration point of view, for an external dev configuring a new platform, a variable that defaults to some value if not provided, regardless of how that resolution occurs, is not required.

Yes, "not required" to be set in `env.local`, but not "not required" for operating the product.  If a var is not required for operating the product, then it would not exist at all.

> It would be even more misleading to say that OPTIONAL_VARS is required.

It's "optional" for `env.local`.  Just to be clear, it has always been about their presence in the `env.local` file.